File: JS/FrotaDeTaxi.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

let lines = gets().split('\n');

const [precoAlcool, precoGasolina, rendimentoAlcool, rendimentoGasolina] = lines[0].split(' ').map(x => parseFloat(x));

print( moreEconomic(precoAlcool, precoGasolina, rendimentoAlcool, rendimentoGasolina) );

/**
 * Return the more economic combustivel.
 */
function moreEconomic(a, g, ra, rg) {
    let result = "A";
    let precoPorKmAlcool = a / ra;
    let precoPorKmGasolina = g / rg;

    if ( precoPorKmGasolina <= precoPorKmAlcool ) result = "G";

    return result;
}
module.exports = {moreEconomic}

File: GO/FrotaDeTaxi.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

/**
 * Return the more economic combustivel.
 */
func moreEconomic(a float64, g float64, ra float64, rg float64) string {
	result := "A"
	precoPorKmAlcool := a / ra
	precoPorKmGasolina := g / rg

	if precoPorKmGasolina <= precoPorKmAlcool {
		result = "G"
	}

	return result
}
